### PR TITLE
Upgrade NeoFS SDK Go library

### LINKED ISSUE #122: Uploaded objects with zero payload instead of expected data
When loading objects via NeoFS.send, periodically (not every time, which complicates reproducibility), an object with a zero payload is created.

Head example:
```
$ neofs-cli ... object head --cid 4Pm2ptHQQ3pq6euM9PnEP3MgPC65NRPmFJd3VPhYqzTi --oid 2koFXqVTV6xHhBJ4hHW2mvs12wmoS19NXSL9vfRXw97x 
ID: 2koFXqVTV6xHhBJ4hHW2mvs12wmoS19NXSL9vfRXw97x
CID: 4Pm2ptHQQ3pq6euM9PnEP3MgPC65NRPmFJd3VPhYqzTi
Owner: NceCu75mcC9pCu9D8ykj1jyZQGjzXcitGZ
CreatedAt: 3088
Size: 0
HomoHash: 00000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001
Checksum: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
Type: regular
Attributes:
  Owner-Email=anatoly@nspcc.ru
  Email=7a7ba8cf152b04e2c39218f4ee331917f9e935113e65ada6304015fe88a24dd8
  __NEOFS__EXPIRATION_EPOCH=3090
  FileName=photo_2021-01-25_12-12-38.jpg
Split Header Signature:
  public key: 031445f5788a237355912f71ec80a7e317250a8c099542a2049ddcf96dbbb33b8d
  signature: 041891b73770ce153826361d0a4d0fd3324ba1ebcdc22cbd8f4b4299d2dc3aaf6ee847365708dad7d029eb6cdb2fc2717f21787277ae93ebbf977a92cf1a142e47
(prod:neofs)
```

### PR DESCRIPTION
**Pull request by cthulhu-rider**
* based on and blocked by https://github.com/nspcc-dev/neofs-sdk-go/pull/135
* closes #122

### CODE REVIEW

**Comment by KirillovDenis** [comment]
I think this PR can close https://github.com/nspcc-dev/neofs-http-gw/issues/122

File: downloader/download.go [hunk 1]
```diff
@@ -325,75 +316,55 @@ func (d *Downloader) byAttribute(c *fasthttp.RequestCtx, f func(request, pool.Ob
...
-		log.Debug("found multiple objects",
-			zap.Strings("object_ids", objectIDs(ids).Slice()),
-			zap.Stringer("show_object_id", ids[0]))
-	}
+	buf := make([]oid.ID, 1)
 
-	return formAddress(cid, ids[0]), nil
-}
+	n, err := res.Read(buf)
+	if n == 0 {
```
Comment (KirillovDenis) [inline downloader/download.go:331]: Do we want to make sure that result contains exactly one id ?

File: downloader/download.go [hunk 2]
```diff
@@ -403,71 +374,110 @@ func (d *Downloader) DownloadZipped(c *fasthttp.RequestCtx) {
...
-			Method:   compression,
-			Modified: time.Now(),
-		}
-		entryWriter, err := zipWriter.CreateHeader(header)
+	optBearer := bearerOpts(c)
+	empty := true
+	n := 0
+	buf := make([]oid.ID, 10) // configure?
+
+iterator:
```
Comment (KirillovDenis) [inline downloader/download.go:410]: We can update sdk to https://github.com/nspcc-dev/neofs-sdk-go/commit/07817fb4032db26e32366a2e06f1e222aa0f9740 and use iterator from pool

File: downloader/download.go (hunk 1 above)
Comment (cthulhu-rider) [reply downloader/download.go:331]: In previous implementation that was needed for debug message only, do u think this is still useful?

File: integration_test.go [hunk 3]
```diff
@@ -271,10 +276,10 @@ func getPool(ctx context.Context, t *testing.T, key *keys.PrivateKey) pool.Pool
...
 
 	opts := &pool.BuilderOptions{
-		Key:                    &key.PrivateKey,
-		NodeConnectionTimeout:  5 * time.Second,
-		NodeRequestTimeout:     5 * time.Second,
-		SessionExpirationEpoch: math.MaxUint64,
+		Key:                       &key.PrivateKey,
+		NodeConnectionTimeout:     5 * time.Second,
+		NodeRequestTimeout:        5 * time.Second,
+		SessionExpirationDuration: math.MaxUint64,
```
Comment (KirillovDenis) [inline integration_test.go:282]: We can drop this parameter

File: downloader/download.go (hunk 1 above)
Comment (KirillovDenis) [reply downloader/download.go:331]: > In previous implementation that was needed for debug message only

Didn't know about this. Then ok

File: downloader/download.go (hunk 2 above)
Comment (cthulhu-rider) [reply downloader/download.go]: https://github.com/nspcc-dev/neofs-http-gw/pull/126/commits/c13a68e81e66efeb32926f98147ee07d70a16838

**Comment by KirillovDenis** [comment]
The last commit message should be `[#126] ...` not `[#346] ...`

**Pull request merged**